README.adoc

```diff
@@ -1,9 +1,135 @@
 = PostgreSQL Fingerprint Storage Plugin
 
-// TODO: add more badges
+link:https://ci.jenkins.io/job/Plugins/job/postgresql-fingerprint-storage-plugin/job/master/[
+image:https://ci.jenkins.io/job/Plugins/job/postgresql-fingerprint-storage-plugin/job/master/badge/icon[Build]]
+link:https://lgtm.com/projects/g/jenkinsci/postgresql-fingerprint-storage-plugin/context:java[
+image:https://img.shields.io/lgtm/grade/java/g/jenkinsci/postgresql-fingerprint-storage-plugin.svg?logo=lgtm&logoWidth=18[Language grade: Java]]
+link:https://github.com/jenkinsci/postgresql-fingerprint-storage-plugin/graphs/contributors[
+image:https://img.shields.io/github/contributors/jenkinsci/postgresql-fingerprint-storage-plugin.svg?color=blue[Contributors]]
+link:https://gitter.im/jenkinsci/external-fingerprint-storage[
+image:https://badges.gitter.im/jenkinsci/external-fingerprint-storage.svg[Gitter]]
 
-This plugin is under development.
-Please refer to https://www.jenkins.io/projects/gsoc/2020/projects/external-fingerprint-storage/[project page]
-for current status.
+image::images/logo.png[]
 
-// TODO: add more content
+== Introduction
+
+The plugin allows users to configure PostgreSQL for storing fingerprints.
+
+Another plugin which supports link:https://github.com/jenkinsci/redis-fingerprint-storage-plugin/[
+storing fingerprints externally, but inside Redis instances] is also available.
+Consider trying it out too! It is much more stable than this plugin at the moment,
+and also supports fingerprint cleanup, which is currently unsupported in this plugin.
+
+The plugin is currently under active development, please refer to the
+link:https://www.jenkins.io/projects/gsoc/2020/projects/external-fingerprint-storage/[project page] for the latest
+status.
+
+== Installation
+
+Clone the repository
+
+```
+git clone https://github.com/jenkinsci/postgresql-fingerprint-storage-plugin
+cd postgresql-fingerprint-storage-plugin
+```
+
+To obtain the `postgresql-fingerprint-storage.hpi` and `postgresql-fingerprint-storage.jar` files, run:
+```
+mvn clean install
+```
+
+They will be available inside `./plugin/target`.
+(Note that docker is required for running tests.
+If Docker is not installed, consider skipping tests)
+
+To run the plugin on a local instance, run:
+
+```
+cd plugin
+mvn hpi:run
+```
+
+== Configuring the plugin
+
+=== Using the Web UI
+
+Once the plugin has been installed, you can configure the PostgreSQL server details by following the steps below:
+
+. Select `Manage Jenkins`
+
+. Select `Configure System`
+
+. Scroll to the section `Fingerprints`
+
+. Select `PostgreSQL Fingerprint Storage` from the dropdown for Fingerprint Storage Engine and fill in the required
+details:
+
++
+
+image::images/config_page.png[]
```

Comment:
it might be better to use `docs/images` so that all documentation is localized in the repo if you eventually need more than one page